src/metrics-gatherer.ts

```diff
@@ -40,90 +41,138 @@ export class MetricsGatherer {
 	}
 
 	private initState() {
-		this.metrics = {
-			gauge: {},
-			counter: {},
-			summary: {},
-			histogram: {},
-		};
-		this.customParams = {};
-		this.descriptions = {};
-		this.existMap = {};
-		this.kinds = {};
+		try {
+			this.metrics = {
+				gauge: {},
+				counter: {},
+				summary: {},
+				histogram: {},
+			};
+			this.customParams = {};
+			this.descriptions = {};
+			this.existMap = {};
+			this.kinds = {};
+			this.internalErrorCount = 0;
+		} catch (e) {
+			this.err(e);
+		}
 	}
 
 	// fetch the description for a metric
 	public describe(name: string, text: string, custom: CustomParams = {}) {
-		if (this.descriptions[name]) {
-			throw new MetricsGathererError(
-				`tried to describe metric "${name}" twice`,
-			);
+		try {
+			if (this.descriptions[name]) {
+				this.err(
+					new MetricsGathererError(`tried to describe metric "${name}" twice`),
+				);
+				return;
+			}
+			this.descriptions[name] = text;
+			this.customParams[name] = custom;
+		} catch (e) {
+			this.err(e);
 		}
-		this.descriptions[name] = text;
-		this.customParams[name] = custom;
 	}
 
 	// observe a gauge metric
 	public gauge(name: string, val: number, labels: LabelSet = {}) {
-		this.ensureExists(name, 'gauge', labels);
-		this.metrics.gauge[name].set(labels, val);
+		try {
+			this.ensureExists(name, 'gauge', labels);
+			this.metrics.gauge[name].set(labels, val);
+		} catch (e) {
+			this.err(e);
+		}
 	}
 
 	// increment a counter or gauge
 	public inc(name: string, val: number = 1, labels: LabelSet = {}) {
-		// ensure either that this metric already exists, or if not, create a gauge
-		this.ensureExists(name, 'gauge', labels);
-		if (!this.checkMetricType(name, ['gauge', 'counter'])) {
-			throw new MetricsGathererError(
-				`Tried to increment non-gauge, non-counter metric ${name}`,
-			);
-		}
-		if (this.kinds[name] === 'gauge') {
-			this.metrics.gauge[name].inc(labels, val);
-		} else {
-			this.metrics.counter[name].inc(labels, val);
+		try {
+			// ensure either that this metric already exists, or if not, create a gauge
+			this.ensureExists(name, 'gauge', labels);
+			if (!this.checkMetricType(name, ['gauge', 'counter'])) {
+				this.err(
+					new MetricsGathererError(
```

Comment:
Might be worth just throwing this and handling it with everything else in the `catch` below